catalog/security/catalog-security-plugin/src/main/java/ddf/catalog/security/plugin/SecurityPlugin.java

```diff
@@ -49,13 +49,23 @@ public SecurityPlugin(SubjectIdentity subjectIdentity) {
     this.subjectIdentity = subjectIdentity;
   }
 
+  /**
+   * Sets the {@link Metacard#POINT_OF_CONTACT} attribute on a {@link Metacard} if it does not
+   * contain one already and the metacard is a resource metacard. From a legacy perspective, a
+   * metacard with no tags is considered a resource metacard.
+   *
+   * @param input the {@link CreateRequest} containing {@link Metacard}s to process
+   * @return the original {@link CreateRequest}
+   */
   @Override
-  public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+  public CreateRequest processPreCreate(CreateRequest input) {
     ddf.security.Subject subject = setSubjectOnRequestProperties(input);
 
     if (input.getMetacards() != null && subject != null) {
       input
           .getMetacards()
+          .stream()
+          .filter(metacard -> metacard.getAttribute(Metacard.POINT_OF_CONTACT) == null)
```

Comment:
Was the original intent of this plugin to make it impossible for someone to create a metacard with a point of contact other than the subject that was creating it? I think you should be able to do this but wonder if we are breaking some assumption/requirement somewhere.

Reply:
This POC functionality was added as leadup to a feature that never got implemented. That feature was to allow the POC of a metacard to see that metacard regardless of their security markings. The purpose being to circumvent the scenario where someone ingests a resource that they can't then query for.

That being said, I don't think this is an issue. The POC of a metacard should definitely be set to the value of the Subject who originally ingested the resource, so if it's already there we shouldn't be modifying it.